internal/controller/admissionpolicy_controller.go

```diff
@@ -101,6 +103,40 @@ func (r *AdmissionPolicyReconciler) findAdmissionPoliciesForPod(ctx context.Cont
 	return findPoliciesForPod(ctx, r.Client, object)
 }
 
-func (r *AdmissionPolicyReconciler) findAdmissionPolicyForWebhookConfiguration(_ context.Context, webhookConfiguration client.Object) []reconcile.Request {
-	return findPolicyForWebhookConfiguration(webhookConfiguration, false, r.Log)
+//nolint:dupl // This function is similar to the one for AdmissionPolicyGroup
+func (r *AdmissionPolicyReconciler) findAdmissionPolicyForWebhookConfiguration(ctx context.Context, webhookConfiguration client.Object) []reconcile.Request {
+	if !hasKubewardenLabel(webhookConfiguration.GetLabels()) {
+		return []reconcile.Request{}
+	}
+
+	policyName, policyNameExists := webhookConfiguration.GetAnnotations()[constants.WebhookConfigurationPolicyNameAnnotationKey]
+	if !policyNameExists {
+		return []reconcile.Request{}
+	}
+
+	policyNamespace, policyNamespaceExists := webhookConfiguration.GetAnnotations()[constants.WebhookConfigurationPolicyNamespaceAnnotationKey]
+	if !policyNamespaceExists {
+		return []reconcile.Request{}
+	}
+
+	var admissionPolicy policiesv1.AdmissionPolicy
+	err := r.Get(ctx, client.ObjectKey{
+		Name:      policyName,
+		Namespace: policyNamespace,
+	}, &admissionPolicy)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			r.Log.Error(err, "cannot retrieve AdmissionPolicy")
+		}
+		return []reconcile.Request{}
+	}
```

Comment:
Why do we need to perform the lookup of the policy? Can't we just fire up the reconciliation event and let the policy reconciliation loop deal with the case of a missing policy?

I know this `Get` request is using the cache of k8s's client, but it seems like a duplication of code

Reply:
you are totally right, I removed the get as we already do this check in the reconciler ignoring the not found error.